relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -199,10 +215,54 @@ fn keep_transaction_name(source: &TransactionSource) -> bool {
     }
 }
 
+/// Decide whether we want to keep the transaction name.
+/// High-cardinality sources are excluded to protect our metrics infrastructure.
+/// Note that this will produce a discrepancy between metrics and raw transaction data.
+fn get_transaction_name(
+    event: &Event,
+    accept_transaction_names: &AcceptTransactionNames,
+) -> Option<String> {
+    let original_transaction_name = match event.transaction.value() {
+        Some(name) => name,
+        None => {
+            return None;
+        }
+    };
+
+    // In client-based mode, handling of "unknown" sources depends on the SDK name.
+    // In strict mode, treat "unknown" as high cardinality.
+    let treat_unknown_as_low_cardinality = matches!(
+        accept_transaction_names,
+        AcceptTransactionNames::ClientBased
+    ) && {
+        let sdk_name = event
+            .client_sdk
+            .value()
+            .and_then(|sdk| sdk.name.value())
+            .map(|s| s.as_str())
+            .unwrap_or_default();
+
+        !["sentry.javascript.browser", "sentry.javascript.react"].contains(&sdk_name)
```

Comment:
_nit_: Let's move this entire thing into a utility function for readability, something like `fn is_browser_sdk(&Event)`.

Reply:
Are we fine with just specific js sdks? I thought a lot of them had high cardinality names, along with some other sdk's like Ruby?

Reply:
We could also turn this around this and specify the SDKs that we trust to send only low-cardinality names. @jan-auer what do you think?

Reply:
It depends on both the prevalence and the absolute volume of high cardinality in these platforms. We could start with just JS and as we increase the sample rate extend it with more SDKs or even specific integrations that cause problems.